Terminal.Gui/Application/Application.cs

```diff
@@ -68,6 +68,14 @@ internal static List<CultureInfo> GetSupportedCultures ()
                       .ToList ();
     }
 
+    /// <summary>
+    ///     Gets the size of the screen. This is the size of the screen as reported by the <see cref="ConsoleDriver"/>.
+    /// </summary>
+    /// <remarks>
+    ///     If the <see cref="ConsoleDriver"/> has not been initialized, this will return a default size of 2048x2048; useful for unit tests.
+    /// </remarks>
+    public static Rectangle Screen => Driver?.Screen ?? new (0, 0, 2048, 2048);
```

Comment:
The primary issue with something that big, of course, is that it's a HUGE rectangle.

Instead of a hard-coded value for the dimensions of the rectangle, maybe the width and height should be pulled from configuration, with sane fallbacks like the current 2048 if not specified?

Reply:
> Maybe a value of short.MaxValue would be better than 2048 for the fallback?
> 
> The reasoning there being that the screen buffer can be up to that large in each dimension.
> 
> Additionally, but out of scope for this PR, we probably should include a few tests that actually try to place things at the limits of the screen buffer and beyond, to ensure those cases are handled properly.

There are cases where TextFormatter can overflow memory with `int.MaxValue`. Maybe `short.MaxValue` would be ok. 

However, this is just to support Unit tests.